src/main/java/org/opensearch/flowframework/workflow/AbstractRetryableWorkflowStep.java

```diff
@@ -96,16 +98,50 @@ protected void retryableGetMlTask(
                                     id,
                                     ActionListener.wrap(updateResponse -> {
                                         logger.info("successfully updated resources created in state index: {}", updateResponse.getIndex());
-                                        future.complete(
-                                            new WorkflowData(
-                                                Map.ofEntries(
-                                                    Map.entry(resourceName, id),
-                                                    Map.entry(REGISTER_MODEL_STATUS, response.getState().name())
-                                                ),
+                                        if (workflowStep == "Local model registration" && Boolean.TRUE.equals(deploy)) {
```

Comment:
Perhaps a simpler way would to pass an action listener as part of the retryableGetMlTask. This would allow us to handle completing the future / simulating model deployment within the step class after the action listener's `onResponse` is invoked.

Reply:
I have added the modifications we discussed in the latest commit